traefik_mesh/tests/test_unit.py

```diff
@@ -0,0 +1,110 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import mock
+import pytest
+
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.dev.utils import assert_service_checks, get_metadata_metrics
+from datadog_checks.traefik_mesh import TraefikMeshCheck
+
+from .common import (
+    OM_METRICS,
+    OM_MOCKED_INSTANCE,
+    OM_MOCKED_INSTANCE_CONTROLLER,
+    OPTIONAL_METRICS,
+    get_fixture_path,
+    read_json_fixture,
+)
+
+
+def test_check_mock_traefik_mesh_openmetrics(dd_run_check, aggregator, mock_http_response):
+    mock_http_response(file_path=get_fixture_path('traefik_proxy.txt'))
+    check = TraefikMeshCheck('traefik_mesh', {}, [OM_MOCKED_INSTANCE])
+    dd_run_check(check)
+
+    for metric in OM_METRICS:
+        if metric in OPTIONAL_METRICS:
+            aggregator.assert_metric(metric, at_least=0)
```

Comment:
Do you mean since we're mocking the `traefik_proxy.txt` as the response payload, we should be able to validate the presence of metrics in `OPTIONAL_METRICS`. The `traefik_proxy.txt` does have some of the metrics available but not all. Since we can't validate the presence of the metrics that aren't there, we set the condition `at_least=0`.

Reply:
ok, so we know for sure that all of `OPTIONAL_METRICS` are missing from the `traefik_proxy.txt` payload?

Reply:
Yep yep